saleor/graphql/discount/tests/benchmark/test_promotion_create.py

```diff
@@ -106,3 +123,83 @@ def test_promotion_create(
     # then
     data = content["data"]["promotionCreate"]
     assert data["promotion"]
+
+
+@pytest.mark.django_db
+@pytest.mark.count_queries(autouse=False)
+def test_promotion_create_order_promotion(
+    staff_api_client,
+    permission_group_manage_discounts,
+    channel_USD,
+    product_variant_list,
+    count_queries,
+    django_assert_num_queries,
+):
+    # given
+    permission_group_manage_discounts.user_set.add(staff_api_client.user)
+    order_predicate_0 = {
+        "discountedObjectPredicate": {"baseSubtotalPrice": {"range": {"gte": "10"}}}
+    }
+    order_predicate_1 = {
+        "discountedObjectPredicate": {"baseSubtotalPrice": {"range": {"gte": "50"}}}
+    }
+    order_predicate_2 = {
+        "discountedObjectPredicate": {"baseSubtotalPrice": {"range": {"gte": "100"}}}
+    }
+    channel_ids = [graphene.Node.to_global_id("Channel", channel_USD.pk)]
+    gift_ids_1 = [
+        graphene.Node.to_global_id("ProductVariant", variant.pk)
+        for variant in product_variant_list[:2]
+    ]
+    gift_ids_2 = [
+        graphene.Node.to_global_id("ProductVariant", variant.pk)
+        for variant in product_variant_list[2:]
+    ]
+
+    variables = {
+        "input": {
+            "name": "test gift promotion",
+            "type": PromotionTypeEnum.ORDER.name,
+            "rules": [
+                {
+                    "name": "test order promotion rule 1",
```

Comment:
Maybe you could make this `rules` list with list comprehension? Then we could use test  `parametrize` which could provide the quantity of the rules that should be included in `rules` list 🤔

Reply:
This is a benchmark test, which counts the number of queries. Checking the logic is secondary.